docroot/modules/humsci/hs_admin/hs_admin.module

```diff
@@ -196,3 +203,13 @@ function hs_admin_field_widget_complete_paragraphs_browser_form_alter(&$field_wi
     }
   }
 }
+
+/**
+ * Implements hook_user_login().
+ */
+function hs_admin_user_login(UserInterface $account) {
+  if ($account->isAuthenticated() && $account->hasPermission('access toolbar')) {
```

Comment:
Instead of setting a state, why not do the access check when adding the library in the `hook_preprocess_node_edit_form`?

Reply:
@cienvaras

Reply:
@joegl We need to execute the JS code only once when the user logs in. It's not possible to add and execute the JS  in `hook_user_login()`, so the only way that we found to do this was to set a temporary state for loading the JS in `hook_page_attachments()` once and then remove it to prevent it from loading again. We considered adding all the logic in Javascript as @pookmish suggested in a previous related comment, but we couldn't find a way to react to login events from javascript.

Reply:
You're already using the local storage in the JS anyways, why not just leverage it a little bit more. ```
if window.localStorage.getItem('do-the-thing') !== "false"
do the thing
window.localStorage.setItem('do-the-thing', false)
```

Reply:
@pookmish @joegl Thanks for the advice, I just pushed the updated code. Moved the logic to the JS code as requested and removed the state.